### PR TITLE
Update shop stock indicator

### PR DESCRIPTION
**Pull request by transcental**

### CODE REVIEW

**Comment by transcental** [comment]
Tested this locally and it works. Thanks to @SkyfallWasTaken for starting this and getting most of it working

**Comment by transcental** [comment]
Out of stock items are shown at the bottom and are greyed out. The stock indicator is based on items that are "Awaiting Fulfillment" and "Fulfilled".

Also moved #1267 into this PR and closed it.

image
image

File: pages/api/arcade/shop.js [hunk 1]
```diff
@@ -1,37 +1,45 @@
...
-    baseID: "app4kCWulfB02bV8Q",
+    baseID,
     tableName: "Shop Items"
   })
   const ordersTable = new AirtablePlus({
     apiKey: process.env.AIRTABLE_API_KEY,
-    baseID: "app4kCWulfB02bV8Q",
+    baseID,
     tableName: "Orders"
   })
```
Comment (skyfallwastaken) [inline pages/api/arcade/shop.js]: We can replace `shopItemsTable` and `ordersTable` with one AirtablePlus class, and pass `tableName` to the read calls to increase perf

File: pages/api/arcade/shop.js [hunk 2]
```diff
@@ -1,37 +1,45 @@
...
+  const newRecordsPromise = records.map(async record => {
     const fields = record.fields;
     let stock = fields["Stock"]
 
-    if (stock) {
-      stock -= (await ordersTable.read({
+    if (stock && fields["Orders"]) {
+      const orderIds = fields["Orders"]
+      const ordersFilter = orderIds.map(id => `RECORD_ID() = "${id}"`).join(", ")
+      const data = await ordersTable.read({
```
Comment (skyfallwastaken) [inline pages/api/arcade/shop.js:24]: (see above)

**Pull request merged**